src/Analyzers/Core/CodeFixes/UseCoalesceExpression/AbstractUseCoalesceExpressionForIfNullStatementCheckCodeFixProvider.cs

```diff
@@ -33,10 +26,17 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
         return Task.CompletedTask;
     }
 
-    protected override Task FixAllAsync(
+    protected abstract bool ShouldAddExplicitCast(
+        ISyntaxFactsService syntaxFacts, SemanticModel semanticModel,
```

Comment:
don't need to pass syntaxfacts in.  since the subclasses are already lang specific, they can directly do whatever they need.

Reply:
I changed the implementation so SyntaxFacts are not passed in, now I am wondering whether I should do the same for the SemanticModel and have the language specific stuff return just true/false for whether the cast can be added?